### PR TITLE
[11.0][ADD] purchase_order_secondary_unit: New module to allow buy in secondary units

### PR DESCRIPTION
**Pull request by sergio-teruel**
This PR depends of https://github.com/OCA/product-attribute/pull/398
cc @Tecnativa

### CODE REVIEW

File: purchase_order_secondary_unit/readme/CONTRIBUTORS.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+* Sergio Teruel <carlos.dauden@tecnativa.com>
```
Comment (lmignon) [inline purchase_order_secondary_unit/readme/CONTRIBUTORS.rst]: s/Sergio Teruel/Carlos Dauten/

File: purchase_order_secondary_unit/tests/test_purchase_order_secondary_unit.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields
+from odoo.tests import SavepointCase
+
+
+class TestPurchaseOrderSecondaryUnit(SavepointCase):
+    at_install = False
```
Comment (lmignon) [inline purchase_order_secondary_unit/tests/test_purchase_order_secondary_unit.py]: PLZ remove `at_install`and `post_install` https://github.com/OCA/odoo-community.org/issues/28

**Review by lmignon — CHANGES_REQUESTED** [review]
Thank you @sergio-teruel Some little comments

**Comment by lmignon** [comment]
@sergio-teruel Can you also check travis. Some dependencies are missing

**Comment by sergio-teruel** [comment]
@lmignon Thanks!!. Changes done. The tests will fails until the https://github.com/OCA/product-attribute/pull/398 has not been merged.

File: purchase_order_secondary_unit/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    'summary': 'Purchase product in a secondary unit',
+    'version': '11.0.1.0.0',
+    'development_status': 'Beta',
+    'category': 'Purchase',
+    'website': 'https://github.com/OCA/purchase-workflow',
+    'author': 'Tecnativa, Odoo Community Association (OCA)',
+    'license': 'AGPL-3',
+    'application': False,
+    'installable': True,
+    'auto_install': True,
```
Comment (tarteo) [inline purchase_order_secondary_unit/__manifest__.py:14]: IMO auto_install should be False for this module.

**Review by tarteo — CHANGES_REQUESTED** [review]
Code LGTM. Couldn't test it, Runbot is not working.

File: purchase_order_secondary_unit/__manifest__.py (hunk 3 above)
Comment (pedrobaeza) [reply purchase_order_secondary_unit/__manifest__.py:14]: If you have installed purchase_secondary_unit and purchase, it's logic that this should be auto-install, isn't it?

Comment (tarteo) [reply purchase_order_secondary_unit/__manifest__.py:14]: I misread the module name I thought it was purchase_secondary_unit, product_secondary_unit makes more sense.

**Review by tarteo — CHANGES_REQUESTED** [review]
Tested the functionality and it works fine:

Can you make the labels consistent
image

**Comment by sergio-teruel** [comment]
@tarteo Are you referring to changing the name form "Secondary uom" to "Secondary product unit of measure"?.. I think that this name is very long.... :astonished:

**Comment by tarteo** [comment]
@sergio-teruel Yes, just so it's consistent with the other label.

**Comment by GSLabIt** [comment]
@tarteo seems not a big issue, secondary uom sound good as well, isn't it?

**Comment by GSLabIt** [comment]
Ready to merge?

File: purchase_order_secondary_unit/models/purchase_order.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        digits=dp.get_precision('Product Unit of Measure'),
+    )
+    secondary_uom_id = fields.Many2one(
+        comodel_name='product.secondary.unit',
+        string='Secondary uom',
+        ondelete='restrict',
+    )
+
+    @api.onchange('secondary_uom_id', 'secondary_uom_qty')
+    def onchange_secondary_uom(self):
```
Comment (rousseldenis) [inline purchase_order_secondary_unit/models/purchase_order.py]: @sergio-teruel Prefix your onchanges by '_'

**Review by rousseldenis — APPROVED** [review]
Minor comment. LGTM

File: purchase_order_secondary_unit/tests/test_purchase_order_secondary_unit.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+                'name': cls.product.name,
+                'product_id': cls.product.id,
+                'product_qty': 1,
+                'product_uom': cls.product.uom_id.id,
+                'price_unit': 1000.00,
+                'date_planned': fields.Datetime.now(),
+            })],
+        })
+        po.onchange_partner_id()
+        cls.order = cls.env['purchase.order'].create(
```
Comment (nikul-serpentcs) [inline purchase_order_secondary_unit/tests/test_purchase_order_secondary_unit.py:47]: @sergio-teruel here  `cls.env['purchase.order']` use multiple time, so use like a global variable.

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

**Review by tarteo — APPROVED** [review]

File: purchase_order_secondary_unit/models/purchase_order.py (hunk 4 above)
Comment (pedrobaeza) [reply purchase_order_secondary_unit/models/purchase_order.py]: @sergio-teruel are you going to change this?

**Review by nikul-serpentcs — APPROVED** [review]
Code Review LGTM :+1:

File: purchase_order_secondary_unit/models/purchase_order.py (hunk 4 above)
Comment (pedrobaeza) [reply purchase_order_secondary_unit/models/purchase_order.py]: I have changed the name to conventions, so I merge.

**Comment by rousseldenis** [comment]
@pedrobaeza @sergio-teruel Did this PR not depending on OCA/product-attribute#398 as announced ?

**Comment by pedrobaeza** [comment]
Well, not technically, but for coherence for not seeing 2 times the same field. Sorry for not checking before. Can we merge the other already?

**Pull request merged**